scripts/generate-types.ts

```diff
@@ -0,0 +1,358 @@
+            return invalidValue(l(ref || "object"), val, key, parent);
+        }
+        const result${anyAnnotation} = {};
+        Object.getOwnPropertyNames(props).forEach(key => {
+            const prop = props[key];
+            const v = Object.prototype.hasOwnProperty.call(val, key) ? val[key] : undefined;
+            result[prop.key] = transform(v, prop.typ, getProps, key, ref);
+        });
+        Object.getOwnPropertyNames(val).forEach(key => {
+            if (!Object.prototype.hasOwnProperty.call(props, key)) {
+                result[key] = ${
+  this._tsFlowOptions.runtimeTypecheckIgnoreUnknownProperties
+    ? 'val[key]'
+    : 'transform(val[key], additional, getProps, key, ref)'
+};
+            }
+        });
+        return result;
+    }
+    if (typ === "any") return val;
+    if (typ === null) {
+        if (val === null) return val;
+        return invalidValue(typ, val, key, parent);
+    }
+    if (typ === false) return invalidValue(typ, val, key, parent);
+    let ref${anyAnnotation} = undefined;
+    while (typeof typ === "object" && typ.ref !== undefined) {
+        ref = typ.ref;
+        typ = typeMap[typ.ref];
+    }
+    if (Array.isArray(typ)) return transformEnum(typ, val);
+    if (typeof typ === "object") {
+        return typ.hasOwnProperty("unionMembers") ? transformUnion(typ.unionMembers, val)
+            : typ.hasOwnProperty("arrayItems")    ? transformArray(typ.arrayItems, val)
+            : typ.hasOwnProperty("props")         ? transformObject(getProps(typ), typ.additional, val)
+            : invalidValue(typ, val, key, parent);
+    }
+    // Numbers can be parsed by Date but shouldn't be.
+    if (typ === Date && typeof val !== "number") return transformDate(val);
+    return transformPrimitive(typ, val);
+}
+
+${this.castFunctionLines[0]} {
+    return transform(val, typ, jsonToJSProps);
+}
+
+${this.castFunctionLines[1]} {
+    return transform(val, typ, jsToJSONProps);
+}
+
+function l(typ${anyAnnotation}) {
+    return { literal: typ };
+}
+
+function a(typ${anyAnnotation}) {
+    return { arrayItems: typ };
+}
+
+function u(...typs${anyArrayAnnotation}) {
+    return { unionMembers: typs };
+}
+
+function o(props${anyArrayAnnotation}, additional${anyAnnotation}) {
+    return { props, additional };
+}
+
+function r(name${stringAnnotation}) {
+    return { ref: name };
+}
+`);
+      this.emitTypeMap();
+    }
+  }
+}
+
+type Schema = {
+  typeName: string;
+  schemaContent: string;
+}
+function generateInterfaces(
+  schemas: Schema[],
+) {
+  const schemaInput = new JSONSchemaInput(undefined);
+  const inputData = new InputData();
+  for (const schema of schemas) {
+    schemaInput.addSource({ name: schema.typeName, schema: schema.schemaContent });
+  }
+  inputData.addInput(schemaInput);
+  return quicktype({
+    inputData,
+    lang: new JsiiCompatTypeScriptTargetLanguage(),
+    rendererOptions: { 'nice-property-names': 'false' },
```

Comment:
We can later add `{ "converters": "all-objects" }` (default is `"top-level"`) if we want to support converters for literally every object in the stack (which we may need later).